packages/edit-navigation/src/components/secondary-sidebar/inserter-sidebar.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { close } from '@wordpress/icons';
+import {
+	__experimentalLibrary as Library,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import {
+	useViewportMatch,
+	__experimentalUseDialog as useDialog,
+} from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editNavigationStore } from '../../store';
+import { useNavigationEditorRootBlock } from '../../hooks';
+
+function InserterSidebar() {
+	const SHOW_PREVIEWS = false;
+
+	const isMobileViewport = useViewportMatch( 'medium', '<' );
+
+	const {
+		navBlockClientId,
+		lastNavBlockItemIndex,
+	} = useNavigationEditorRootBlock();
+
+	const { hasInserterItems, selectedBlockClientId } = useSelect(
+		( select ) => {
+			return {
+				hasInserterItems: select( blockEditorStore ).hasInserterItems(
+					navBlockClientId
+				),
+				selectedBlockClientId: select(
+					blockEditorStore
+				).getSelectedBlock()?.clientId,
+			};
+		},
+		[]
+	);
+
+	const { setIsInserterOpened } = useDispatch( editNavigationStore );
+
+	const [ inserterDialogRef, inserterDialogProps ] = useDialog( {
+		onClose: () => setIsInserterOpened( false ),
+	} );
+
+	// Only concerned with whether there are items to display. If not then
+	// we shouldn't render.
+	if ( ! hasInserterItems ) {
+		return null;
+	}
+
+	const isNavBlockSelected = navBlockClientId === selectedBlockClientId;
+
+	return (
+		<div
+			ref={ inserterDialogRef }
+			{ ...inserterDialogProps }
+			className="edit-navigation-layout__inserter-panel"
+		>
+			<div className="edit-navigation-layout__inserter-panel-header">
+				<Button
+					icon={ close }
+					onClick={ () => setIsInserterOpened( false ) }
+				/>
+			</div>
+			<div className="edit-navigation-layout__inserter-panel-content">
+				<Library
+					shouldFocusBlock={ isMobileViewport }
+					rootClientId={ navBlockClientId }
+					// If the root Nav block is selected then any items inserted by the
+					// global inserter should append after the last nav item. Otherwise
+					// simply allow default Gutenberg behaviour.
+					__experimentalInsertionIndex={
+						isNavBlockSelected ? lastNavBlockItemIndex : undefined
+					}
```

Comment:
I think there are two cases where the end of the nav block should be inserted into. That's when the nav block is selected or when no blocks are selected. In both of those situations the default inserter behavior would be to insert after the nav block, which is not possible because that's a locked template.

Always providing `rootClientId` doesn't feel right to me because it's like we're telling `Library` to always insert in the nav block's root level, which is not the case when inserting into submenus. I'm a bit surprised it works, but I think it could be avoided for safety.

Something like this could make it clearer:
```jsx
const shouldInsertInNavBlock = ! selectedBlockClientId || navBlockClientId === selectedBlockClientId;
// ...
<Library
    shouldFocusBlock={ isMobileViewport }
    rootClientId={ shouldInsertInNavBlock ? navBlockClientId : undefined }
    __experimentalInsertionIndex={
        shouldInsertInNavBlock ? lastNavBlockItemIndex : undefined
    }
 ```

Reply:
Weirdly this is almost exactly what I had locally previously but I reverted it because I found it worked without it (ie: what you see in the PR now). I was also surprised it worked!

Let's be safe and do what you suggested.

Reply:
This is handy 

https://github.com/WordPress/gutenberg/blob/c5758d3fd125ffe14f00a25ba5b1c8080dfc33e4/packages/block-editor/src/components/inserter/hooks/use-insertion-point.js#L20-L32

Reply:
✅